zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -104,45 +127,35 @@ func (ob *Observer) ObserveInbound(ctx context.Context) error {
 	}
 	ob.WithLastBlock(lastBlock)
 
-	// skip if current height is too low
-	if lastBlock < ob.GetChainParams().ConfirmationCount {
-		return fmt.Errorf("observeInboundBTC: skipping observer, current block number %d is too low", currentBlock)
-	}
-
-	// skip if no new block is confirmed
-	lastScanned := ob.LastBlockScanned()
-	if lastScanned >= lastBlock-ob.GetChainParams().ConfirmationCount {
+	// check confirmation
+	blockNumber := ob.LastBlockScanned() + 1
+	if !ob.IsBlockConfirmed(blockNumber) {
 		return nil
 	}
 
 	// query incoming gas asset to TSS address
-	blockNumber := lastScanned + 1
 	// #nosec G115 always in range
 	res, err := ob.GetBlockByNumberCached(int64(blockNumber))
 	if err != nil {
 		ob.logger.Inbound.Error().Err(err).Msgf("observeInboundBTC: error getting bitcoin block %d", blockNumber)
 		return err
 	}
-	ob.logger.Inbound.Info().Msgf("observeInboundBTC: block %d has %d txs, current block %d, last block %d",
-		blockNumber, len(res.Block.Tx), currentBlock, lastScanned)
+	ob.logger.Inbound.Info().Msgf("observeInboundBTC: block %d has %d txs, current block %d, last scanned %d",
+		blockNumber, len(res.Block.Tx), currentBlock, blockNumber-1)
```

Comment:
I would keep this log text in log message. If we treat these values as event and attributes, they won't show in the DD 'CONTENT' column. Clicking on each logs to find out inner attributes makes debugging difficult during incident.
![image](https://github.com/user-attachments/assets/6cac9f1d-6024-476b-9bac-649146371b3d)
![image](https://github.com/user-attachments/assets/dc4ea6e3-7e0a-49d0-9bea-4fa223b42c8d)

Reply:
You can add the fields as columns pretty easily though:

<img width="1740" alt="image" src="https://github.com/user-attachments/assets/b7499483-009e-48e9-8e27-8f0336200b6d">

but imo log refactoring should be done in a separate PR

Reply:
If log fields can be shown in `CONTENT`, we could consider wrapping the generalized fields in `fields.go` (and event move) in attributes.
But imo treating logs like emitting events (e.g. cosmos event) might deviate somewhat from the general purpose of logging, which is primarily for debugging.

Reply:
I agree with Alex on the fact that if we want to consider refactoring, it should be a separate PR so that all logs have the same format .
I think adding attributes would be helpful in log filtering , In my opinion.